haskell/private/actions/compile.bzl

```diff
@@ -104,7 +104,7 @@ def _process_hsc_file(hs, cc, hsc_flags, hsc_inputs, hsc_file):
 
     return hs_out, idir
 
-def _compilation_defaults(hs, cc, java, posix, dep_info, plugin_dep_info, srcs, module_map, import_dir_map, extra_srcs, user_compile_flags, output_mode, with_profiling, interfaces_dir, objects_dir, my_pkg_id, version, plugins, non_default_plugins, preprocessors):
+def _compilation_defaults(hs, cc, java, posix, dep_info, plugin_dep_info, srcs, module_map, import_dir_map, extra_srcs, user_compile_flags, output_mode, with_profiling, interfaces_dir, objects_dir, my_pkg_id, version, extra_ldflags_file, plugins, non_default_plugins, preprocessors):
```

Comment:
`compile_binary` has 22 parameters. `compile_library` has 20 parameters. `_compilation_defaults` has 21.

I propose to list the parameters one per line in `_compilation_defaults` :)

Probably we can make the function more readable not by reducing the parameters, but by refactoring it so it calls smaller functions instead of having a body of 280 lines as it is today.

Reply:
> I propose to list the parameters one per line in _compilation_defaults :)

It's a start :)

> Probably we can make the function more readable not by reducing the parameters, but by refactoring it so it calls smaller functions instead of having a body of 280 lines as it is today.

Yes, that sounds like a good approach. Though better kept in a separate PR.

Reply:
I aligned the parameters vertically.

> Yes, that sounds like a good approach. Though better kept in a separate PR.

I created [this issue](https://github.com/tweag/rules_haskell/issues/1719) to track it.